readthedocs/builds/models.py

```diff
@@ -716,6 +717,10 @@ def config(self):
         Build object (it could be stored in this object or one of the previous
         ones).
         """
+        # TODO: now that we are using a proper JSONField here, we could
+        # probably change this field to be a ForeignKey to avoid repeating the
+        # config file over and over again and re-use them to save db data as
+        # well
```

Comment:
Yea, an FK seems reasonable but would be another field right? I agree that our current implementation is just a hacky FK :)

Reply:
Yeah. I'm thinking about having something like

```
# builds.models.Build
config = ForeignKey(BuildConfig)

# config.models.BuildConfig
data = JSONField(default=None)
```

... and later in the code, I'm expecting to do:

```
build.config = BuildConfig.objects.get_or_create(data=user_config)
```

and relying on Postgres to do the work about deciding if it's the same or a new one properly  😄